api_handlers/zotero.php

```diff
@@ -25,6 +25,49 @@ function query_url_api($ids, $templates) {
        if ($template->has('ssrn'))      expand_by_zotero($template, 'https://papers.ssrn.com/sol3/papers.cfm?abstract_id=' . $template->get('ssrn'));
        if ($template->has('doi') && !doi_active($template->get('doi')))  expand_by_zotero($template, 'https://dx.doi.org/' . urlencode($template->get('doi'))); // Non-crossref DOIs, such as 10.13140/RG.2.1.1002.9609
   }
+   
+  // Now that we have expanded URLs, try to lose them
+  $ch = curl_init();
+  curl_setopt($ch, CURLOPT_FOLLOWLOCATION, TRUE);
+  curl_setopt($ch, CURLOPT_MAXREDIRS, 100); // No infinite loops for us
+  curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 4); 
+  curl_setopt($ch, CURLOPT_TIMEOUT, 15);
+  curl_setopt($ch, CURLOPT_RETURNTRANSFER, TRUE);
+  foreach ($templates as $template) {
+    $doi = $template->get_without_comments_and_placeholders('doi');
+    $url = $template->get('url');
+    if ($doi &&
+        $url &&
+        !$template->incomplete() &&
+        !preg_match(REGEXP_DOI_ISSN_ONLY, $doi) &&
+        str_ireplace(CANONICAL_PUBLISHER_URLS, '', $url) != $url &&
+        str_ireplace(['pdf', 'image', 'plate', 'figure', 'picture'], '', $url) == $url &&
+        $template->blank(DOI_BROKEN_ALIASES))
+    {
+          curl_setopt($ch, CURLOPT_URL, "https://dx.doi.org/" . urlencode($doi));
```

Comment:
Why do you even make an actual GET with curl, rather than a simple HEAD request? If your objective is to follow redirects made by the publishers' DOI resolvers, this is impossible for many of them unless you also enable cookies. This may not be the default on all versions of curl.
https://curl.haxx.se/docs/http-cookies.html

A test may be useful to check whether you're getting the dreaded cookieAbsent page, e.g.:
https://www.journals.uchicago.edu/action/cookieAbsent
https://www.tandfonline.com/action/cookieAbsent
https://www.annualreviews.org/action/cookieAbsent
https://aip.scitation.org/action/cookieAbsent